### PR TITLE
cmake: Fixed the sample cdc_acm

### PR DESCRIPTION
**Pull request by SebastianBoe**
Signed-off-by: Sebastian Boe <sebastian.boe@nordicsemi.no>

### CODE REVIEW

**Comment by ydamigos** [comment]
This patch allows to build cdc_acm using cmake but now we define CONFIG_USB_DEVICE_VID and CONFIG_USB_DEVICE_PID in two files prj.conf and sample.yaml.
It would be nice if we could set the options without default values during the configuration, like we are able to do with make which asks for user input for the options without default values.
If we use this solution all subsys/usb samples need this patch.

**Comment by SebastianBoe** [comment]
"It would be nice if we could set the options without default values during the configuration"

I don't understand that this is desirable from a user perspective.

As a user, if I don't know what VID or PID is I don't care what they are either and want sane default values. If I do know what they are I will reconfigure.

If you agree, I suggest we patch all of subsys/usb with this patch.

**Comment by jfischer-phytec-iot** [comment]
>"It would be nice if we could set the options without default values during the configuration"

>I don't understand that this is desirable from a user perspective.

Current status is that the user is explicitly advised that there is no correct VID / PID.

**Comment by SebastianBoe** [comment]
> Current status is that the user is explicitly advised that there is no correct VID / PID.

I understand. Can we use another mechanism to explicitly advise the user about this?

Perhaps one of these:

preprocessor warning
cmake build script warning
FW printk
comment in .conf file
comment in build scripts

**Comment by jfischer-phytec-iot** [comment]
I would suggest `cmake build script warning && comment in .conf file`. @finikorg ?

**Review by finikorg — DISMISSED** [review]
I think we decided not to put explicit values to prj.conf, they are allowed only to sample.yaml for sanity check. @pfl

**Comment by SebastianBoe** [comment]
Yeah, but I am saying that doesn't make sense from a user perspective.

If you are just testing USB then you don't care about PID or VID and just want dummy values.

If we document and warn the user about these dummy values then he will reconfigure long before the product is released.

**Comment by SebastianBoe** [comment]
@jfischer-phytec-iot : I organized it slightly differently to avoid copy-paste. But I can remove the Kconfig default and copy-paste into every prj.conf if you think this is best.

File: samples/subsys/usb/cdc_acm/prj.conf [hunk 1]
```diff
@@ -15,5 +15,9 @@ CONFIG_USB_COMPOSITE_DEVICE=n
 
 CONFIG_USB_MASS_STORAGE=n
 CONFIG_SYS_LOG_USB_MASS_STORAGE_LEVEL=1
+
+CONFIG_USB_DEVICE_VID=0xDEAD
```
Comment (jfischer-phytec-iot) [inline samples/subsys/usb/cdc_acm/prj.conf]: This should not be necessary (L19,L20) because you have set default in `subsys/usb/Kconfig`.

**Review by jfischer-no — COMMENTED** [review]
If a dialogue like we had in Kbuild is no longer possible then I am in favor of this solution. I can not test now, but usb samples built for tinytile should not show the warning because this board has a VID / PID.

**Comment by SebastianBoe** [comment]
"If a dialogue like we had in Kbuild is no longer possible"
It is not possible unfortunately, perhaps it will be in the future, but I have not yet seen a good enough use-case to warrant spending time on this feature.

File: samples/subsys/usb/cdc_acm/prj.conf (hunk 1 above)
Comment (SebastianBoe) [reply samples/subsys/usb/cdc_acm/prj.conf]: Fixed.

**Comment by SebastianBoe** [comment]
>  I can not test now, but usb samples built for tinytile should not show the warning because this board has a VID / PID.

I have verified that tinytile does not trigger the warning but other boards do.

**Review by ydamigos — APPROVED** [review]

**Pull request merged**